### PR TITLE
feat(training): (BAB-71) - add social narrative rewards for non-trader archetypes 

### PR DESCRIPTION
**Pull request by revlentless**
## Summary

- **Social Butterfly and other non-trading archetypes can now achieve high scores without trading profitably.** The reward function previously over-weighted PnL, making it impossible for social-focused agents to succeed.
- Added `calculate_social_reward()` with four components: engagement, information spread, network building, and narrative alignment—each with archetype-specific weights.
- Integration test proves a Social Butterfly with zero P&L but strong social metrics outscores a passive trader.

## Type

- [x] Feature
- [ ] Bug fix
- [ ] Refactor / cleanup (no behavior change)
- [ ] Performance
- [ ] Infra / ops
- [x] Docs
- [ ] Contracts / on-chain
- [ ] Other: …

## Context / Links

- Linear: BAB-71 (Social Narrative Rewards)
- Related: Enables training non-trader archetypes like Social Butterfly, Information Trader

## Scope (keep it focused)

- [x] This PR is focused on a single change/theme (not a catch‑all)
- [x] Drive‑by refactors are excluded or split into a separate PR
- [x] Non‑goals / follow‑ups are listed below (with links)

**Non-goals / follow-ups:**
- [ ] Using `social_only_composite_reward()` as the default reward for social archetypes (requires training config changes)
- [ ] Extending narrative alignment to use full timeline data (currently uses prediction accuracy as proxy)

**Areas touched**
- [ ] Web UI (`apps/web`)
- [ ] Web API routes / SSE / A2A (`apps/web`)
- [ ] CLI (`apps/cli`)
- [ ] Docs site (`apps/docs`) / vendor docs (`docs/vendors/*`)
- [ ] Domain / game engine (`packages/engine`, `packages/core/*`)
- [ ] Agents / runtime / A2A / MCP (`packages/agents`, `packages/a2a`, `packages/mcp`)
- [ ] API infra (`packages/api`)
- [ ] DB (`packages/db`)
- [ ] Contracts / on-chain (`packages/contracts`)
- [ ] Shared types/utils (`packages/shared`)
- [ ] Tests (`packages/testing`)
- [x] Other: `packages/training` (Python reward functions, W&B logging, documentation)

## Changes

**New in `rewards.py`:**
- `SocialRewardResult` dataclass - breakdown of social reward components
- `NarrativeEvent` dataclass - ground truth events for narrative alignment
- `calculate_engagement_score()` - posts, DMs, comments, group activity
- `calculate_information_spread_score()` - reactions, shares, follower gains
- `calculate_network_score()` - unique connections, groups, reputation
- `calculate_narrative_alignment_score()` - alignment with revealed events
- `calculate_social_reward()` - combines components with archetype-specific weights
- `social_only_composite_reward()` - PnL-independent composite for social archetypes
- `SOCIAL_REWARD_WEIGHTS` - per-archetype component weights (6 archetypes + default)
- `SOCIAL_COMPOSITE_WEIGHTS` - composite weights for social-focused archetypes
- `_interpolate_score()` helper - reduces redundant threshold logic

**Updated in `babylon_env.py`:**
- Calls `calculate_social_reward()` and stores component scores
- Logs 5 new W&B metrics: `train/social_{engagement,spread,network,narrative,total}_mean`

**New tests (`test_social_rewards.py`):**
- 80 tests covering all social reward functions
- Boundary conditions, edge cases, error handling
- Integration test proving non-trader can outscore passive trader

**Documentation updates:**
- `enhanced-rewards.md` - Social & Narrative Rewards section
- `archetype-system.md` - Social Archetypes section with weight tables
- `adding-archetypes.md` - Guide for adding social archetypes
- `reward-system.md` - Social Rewards section
- `rubrics.md` - Social Butterfly scoring notes
- `glossary.md` - Social Reward, Narrative Event definitions
- `wandb.md` - Social metrics documentation

## Review guide

**Start here**
- `packages/training/python/src/training/rewards.py` lines 1760-2095 (all new social reward code)
- `packages/training/python/tests/test_social_rewards.py` (comprehensive tests)

**Risk**
- [x] Low
- [ ] Medium
- [ ] High

**Notes for reviewers**
- All new code is additive—existing reward functions unchanged
- `social_only_composite_reward()` is not yet wired as default for any archetype (opt-in only)
- Weight values (e.g., Social Butterfly 40% network) are initial estimates; may need tuning after training runs
- Narrative alignment currently uses `prediction_accuracy` as proxy when full timeline data unavailable

## Test plan

**Commands run**
- [x] `bun run check` (Biome format)
- [x] `bun run typecheck`
- [x] `bun run lint`
- [x] `bun run build`
- [x] `bun run test` (unit + integration) - `make tier1` in packages/training
- [ ] `bun run test:e2e` (if critical flows changed)
- [ ] `bun run contracts:test` (if contracts changed)
- [ ] `bun run db:check` (if DB schema/queries changed)

**Manual verification**
1. Ran `pytest tests/test_social_rewards.py -v` - 80 tests passed
2. Ran `make tier1` - 766 tests passed
3. Verified W&B metrics appear in `wandb_log()` output

## Ops / Migration / Deployment

- [x] No deploy impact
- [ ] Requires env var updates (listed below + `.env.example` updated)
- [ ] Requires DB migration (`bun run db:migrate`) / backfill / seed
- [ ] Changes cron schedule or endpoints (`vercel.json`, `CRON_SECRET`, etc.)
- [ ] Rollout behind a flag / gradual rollout

## Breaking changes

- [x] None
- [ ] Yes (describe + migration guide below)

## Security / privacy

- [x] No security impact
- [ ] Needs extra review (describe)

### Option B: No visual changes

- Reason: Backend-only change to Python reward functions. No UI impact. New W&B metrics will appear in training dashboards automatically.

## Checklist (author)
- [x] Self-review done (diff + critical paths)
- [x] Base branch is correct (`staging` by default)
- [x] Handlers remain thin and portable (validate → service → map errors)
- [x] Domain logic stays in packages (no Next/React/Elysia coupling in core)
- [ ] `.env.example` updated (if env changed) and variables documented above
- [x] Docs updated (and `bun run docs:generate` if needed)
- [x] Tests added/updated for behavior changes (or rationale provided)
- [ ] Dependency changes are intentional (`bun.lock` updated)
- [ ] Code owners requested (auto via CODEOWNERS or manual)
- [x] **Screenshots/recordings section filled** (visual demo OR explanation why N/A)

## Summary by CodeRabbit

* **New Features**
  * Social reward metrics system tracking engagement, information spread, network strength, and narrative alignment
  * Counterfactual alpha metric for causal evaluation
  * Narrative event framework for decision tracking

* **Documentation**
  * Expanded guides on social archetypes, weighting, metrics, rubrics, and W&B logging
  * New glossary entries for Social Reward and Narrative Event

* **Tests**
  * Comprehensive test suite covering social reward calculations, weighting validation, and integration scenarios

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<h3>Greptile Summary</h3>

This PR successfully implements a comprehensive social reward system that enables non-trading archetypes (Social Butterfly, Information Trader, etc.) to achieve high scores without profitable trading. Previously, PnL-focused rewards made it impossible for social-focused agents to succeed.

**Key Changes:**
- Adds `calculate_social_reward()` with four weighted components: engagement (social activity volume/diversity), information spread (content reach), network building (connections/reputation), and narrative alignment (prediction accuracy or timeline analysis)
- Each archetype has custom weights (e.g., Social Butterfly prioritizes network at 40%, Information Trader prioritizes narrative at 40%)
- New `social_only_composite_reward()` function minimizes PnL weight (10%) and maximizes social weight (40-55%) for social archetypes
- Integration test proves a Social Butterfly with zero P&L but strong social metrics outscores a passive trader
- Comprehensive test suite with 80 tests covering all functions, boundaries, edge cases, and weight validation
- Full documentation across 7 book chapters with tables, examples, and W&B metric definitions
- W&B logging tracks 5 new metrics: engagement, spread, network, narrative, and total social scores

**Code Quality:**
- Clean implementation using dataclasses (`SocialRewardResult`, `NarrativeEvent`)
- DRY helper function `_interpolate_score()` reduces threshold logic duplication
- All weights validated to sum to 1.0
- Proper boundary handling (clamping, zero-division protection)
- Additive changes only—existing reward functions untouched

**Minor Issue:**
- One inline import should be moved to top-level (style comment added)

<h3>Confidence Score: 4/5</h3>

- This PR is safe to merge with minimal risk—all changes are additive, well-tested, and properly documented.
- Score reflects excellent test coverage (80 tests with boundary/edge cases), comprehensive documentation, clean additive implementation with no changes to existing functionality, and successful integration test. Minor point deduction for one inline import style issue.
- No files require special attention—the minor style issue in `babylon_env.py` is non-critical.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| packages/training/python/src/training/rewards.py | Adds comprehensive social reward system with 4 scoring components (engagement, spread, network, narrative) and archetype-specific weights. Clean implementation with proper dataclasses and helper functions. |
| packages/training/python/src/training/babylon_env.py | Integrates social rewards into training loop, adds W&B logging for 5 social metrics. Uses inline import instead of top-level import (minor style issue). |
| packages/training/python/tests/test_social_rewards.py | Excellent test coverage with 80 tests covering all functions, boundary conditions, edge cases, weight validation, and integration scenarios proving non-traders can outscore traders. |
| packages/training/book/src/scoring/enhanced-rewards.md | Clear documentation of social reward system with component descriptions, archetype weight tables, usage examples, and W&B metrics. |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Env as BabylonEnv
    participant Rewards as rewards.py
    participant Metrics as BehaviorMetrics
    participant WandB as Weights & Biases
    
    Note over Env: Training loop processes trajectory
    
    Env->>Metrics: Extract behavior metrics from trajectory
    Metrics-->>Env: BehaviorMetrics (posts, DMs, connections, etc.)
    
    Env->>Rewards: calculate_social_reward(metrics, archetype)
    
    activate Rewards
    Rewards->>Rewards: calculate_engagement_score(metrics)
    Note right of Rewards: Posts + DMs + comments + groups<br/>Diversity bonus for activity types
    
    Rewards->>Rewards: calculate_information_spread_score(metrics)
    Note right of Rewards: Spread count + reactions + followers<br/>Content reach and influence
    
    Rewards->>Rewards: calculate_network_score(metrics)
    Note right of Rewards: Unique users + groups<br/>Reputation modifier
    
    Rewards->>Rewards: calculate_narrative_alignment_score(metrics)
    Note right of Rewards: Prediction accuracy (proxy)<br/>Or timeline analysis if available
    
    Rewards->>Rewards: Apply archetype-specific weights
    Note right of Rewards: e.g. Social Butterfly:<br/>network=40%, engagement=30%
    
    Rewards-->>Env: SocialRewardResult (components + total)
    deactivate Rewards
    
    Env->>Env: Store component scores in enhanced_reward_metrics
    Note over Env: Accumulate for W&B logging
    
    Env->>WandB: Log social metrics (every N steps)
    Note right of WandB: train/social_engagement_mean<br/>train/social_spread_mean<br/>train/social_network_mean<br/>train/social_narrative_mean<br/>train/social_total_mean

```

### CODE REVIEW

**Review by SYMBaiEX — APPROVED** [review]
## Approval

**LGTM** ✅

Clean, well-tested implementation of social rewards for non-trading archetypes. 

**Highlights:**
- Comprehensive test coverage (80 tests) with boundary conditions and edge cases
- All changes additive—existing reward functions untouched
- Weight validation at module load prevents configuration errors
- Integration test proves Social Butterfly with $0 PnL but strong social metrics outscores passive trader
- Good documentation across 7 book chapters

**Minor suggestions (non-blocking, can address in follow-up):**
- Consider moving inline import (`from .rewards import calculate_social_reward`) to top-level in babylon_env.py
- The 5-tick reaction window in `calculate_narrative_alignment_score()` could be made configurable

Ready to ship.
